src/scattdata.cpp

```diff
@@ -110,7 +113,19 @@ ScattData::base_combine(size_t max_order,
 
   // Combine mult_numer and mult_denom into the combined multiplicity matrix
   xt::xtensor<double, 2> this_mult({groups, groups}, 1.);
-  this_mult = xt::nan_to_num(mult_numer / mult_denom);
```

Comment:
What problem did you observe with the xt::nan_to_num(mult_numer / mult_denom) approach that the explicit loop is intended to fix?

Reply:
It is more complicated issue but the reasaon is the same:

> we didn't save a value of scatter_matrix after first cycle

and it relates `this_mult` array too. Because when we at cross point our last coefficient is 0 and we should divide on 
`this_mult = xt::nan_to_num(mult_numer / mult_denom);` null vector. Function `xt::nan_to_num` converts all NaN to zero leading to when throug simmulation we sample a particle at code line 788 this source
  ```
// Update the weight to reflect neutron multiplicity
  wgt *= mult[gin][i_gout];
```
we always get a "0" is a weight.